stable/grafana/templates/dashboards-json-configmap.yaml

```diff
@@ -12,11 +13,11 @@ metadata:
     heritage: {{ $.Release.Service }}
     dashboard-provider: {{ $provider }}
 data:
-  {{- range $key, $value := $dashboards }}
-    {{- if hasKey $value "json" }}
-  {{ $key }}.json: |
-{{ $value.json | indent 4 }}
-    {{- end }}
-  {{- end }}
-  {{- end }}
+{{- range $key, $value := $dashboards }}
+{{- if hasKey $value "json" }}
+{{ print $key | indent 2 }}.json:
+{{ toYaml ( $files.Get $value.json ) | indent 4}}
```

Comment:
Doesn't this not work when consuming Grafana as a subchart via requirements.yaml?

From https://docs.helm.sh/chart_template_guide/#lines:

![image](https://user-images.githubusercontent.com/20929870/51349070-58cdff80-1a72-11e9-9a45-7da5b2d6dcb5.png)

Reply:
You are probably right.
Would you say that both methods are needed?

Reply:
I think the old approach is definitely needed to support ‘standard’ deployments that purely configure this chart via values instead of modifying its actual contents before deployment.  As for the new approach, the change made here basically requires that consumers have a local copy of the source that they then modify to add JSON dashboards before deployment, correct? That seems like an anti-pattern to me - especially given the fact that it is mostly straightforward to use Helm’s file helpers to load files into a ConfigMap and then pass the name of the ConfigMap as a value to this chart. This would become even easier with the addition of https://github.com/helm/charts/pull/10670. At that point, consumers could even more easily provide actual JSON files without having to keep a local copy of this chart

Reply:
I understand :)
You make some compelling arguments about the raw json option, and I think that json files are also required, but not at the cost of changing expected behavior.

What I would do is:
Return the chart to its previous state with `json`, and add an additional `file` option for folks who keep the chart in their own repos (i think there are more then a few).

What do you think?

In case you agree, and since I am without access to a computer until february, will you be willing to take this on? I would approve this with priority.

Reply:
I suppose I'm a bit biased since my organization prefers to use Helm's dependency management features instead of keeping local copies of charts and modifying them to enable easier upgrades with fewer conflicts, etc. What do you think about reverting the chart to its previous state and then accepting the changes proposed in https://github.com/helm/charts/pull/10670? Then, consumers who create charts that pull in Grafana as a dependency could create a ConfigMap in their 'wrapper' chart and then pass the name of that ConfigMap, which is possibly named according to template functions, to the Grafana subchart. Consumers who instead keep a locally modified copy of the chart could just do the same thing

Reply:
Its not one or the other :)
I'm fine with 10670, but it is not mergable atm.
I prefer not to revert, but to fix.

Reply:
Took some time, but it is now fixed :)